src/Transport/Receiving/MessagePump.cs

```diff
@@ -94,7 +94,20 @@ public async Task StartReceive(CancellationToken cancellationToken = default)
 
             messageProcessingCancellationTokenSource = new CancellationTokenSource();
 
-            circuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker($"'{receiveSettings.ReceiveAddress}'", transportSettings.TimeToWaitBeforeTriggeringCircuitBreaker, ex => criticalErrorAction("Failed to receive message from Azure Service Bus.", ex, messageProcessingCancellationTokenSource.Token));
+            circuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker($"'{receiveSettings.ReceiveAddress}'",
+                transportSettings.TimeToWaitBeforeTriggeringCircuitBreaker, ex =>
+                {
+                    criticalErrorAction("Failed to receive message from Azure Service Bus.", ex,
+                        messageProcessingCancellationTokenSource.Token);
+                }, () =>
+                {
+                    //We don't have to update the prefetch count since we are failing to receive anyway
+                    processor.UpdateConcurrency(1);
```

Comment:
I was wondering if there is some overlap with changes from [#761](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/761/files#diff-bf785c1b37e39481e89856ea4195ca5c7e21f862b8eb2ffc43b63208a3c582e5R207) but I think we are good.